as-controller-board-status/functions/output.go

```diff
@@ -246,15 +248,49 @@ func (boardStatus *CheckBoardStatus) processVendingDoorState(lc logger.LoggingCl
 			return fmt.Errorf("failed to submit the controller board's status to the central vending state service: %v", err.Error())
 		}
 
-		// Prepare a message to be sent to the MQTT bus. Depending on the state
-		// of the door, this message may trigger a CV inference
-		err = boardStatus.RESTCommandJSON(boardStatus.Configuration.DoorStatusCommandEndpoint, http.MethodPut, VendingDoorStatus{
-			VendingDoorStatus: strconv.FormatBool(doorClosed),
-		})
+		// Prepare and send EdgeX command. Depending on the state of the door, this message may trigger a CV inference
+		settings := make(map[string]string)
+		settings["VendingDoorStatus"] = strconv.FormatBool(doorClosed)
+		err = boardStatus.SendCommand(lc, http.MethodPut, boardStatus.Configuration.InferenceDeviceName, boardStatus.Configuration.InferenceDoorStatusCmd,
+			settings)
 		if err != nil {
-			return fmt.Errorf("failed to submit the vending door state to the MQTT device service: %v", err.Error())
+			return fmt.Errorf("failed to submit the vending door state to the command client: %v", err.Error())
 		}
 	}
 
 	return nil
 }
+
+// SendCommand issues CommandClient GET and SET command calls, CommandClient takes care of http calls,
+// here the requirement are actionName, deviceName, commandName and settings, logger client is needed for logging
+func (boardStatus *CheckBoardStatus) SendCommand(lc logger.LoggingClient, actionName string, deviceName string,
+	commandName string, settings map[string]string) error {
+	lc.Debug("Sending Command")
+
+	switch actionName {
+	case http.MethodPut:
+		lc.Debugf("executing %s action", actionName)
+		lc.Debugf("Issuing SET command '%s' for device '%s'", commandName, deviceName)
+
+		response, err := boardStatus.CommandClient.IssueSetCommandByName(context.Background(), deviceName, commandName, settings)
+		if err != nil {
+			return fmt.Errorf("failed to issue '%s' set command to '%s' device: %s", commandName, deviceName, err.Error())
+		}
+
+		lc.Debugf("response status: %d", response.StatusCode)
+
+	case http.MethodGet:
+		lc.Debugf("executing %s action", actionName)
+		lc.Debugf("Issuing GET command '%s' for device '%s'", commandName, deviceName)
+		response, err := boardStatus.CommandClient.IssueGetCommandByName(context.Background(), deviceName, commandName, false, true)
+		if err != nil {
+			return fmt.Errorf("failed to issue '%s' get command to '%s' device: %s", commandName, deviceName, err.Error())
+		}
+		lc.Debugf("response status: %d", response.StatusCode)
+
+	default:
+		return errors.New("Invalid action requested: " + actionName)
+	}
+
+	return nil
```

Comment:
so do we care about the status code is not 200? (like >=400 ) when we return?

Reply:
@jim-wang-intel , That is captured by the Command Client, so we just care if `err` is nil or not.

Reply:
ok, cool thanks